src/isolate.rs

```diff
@@ -567,6 +569,20 @@ impl Isolate {
     );
   }
 
+  fn new_impl(params: CreateParams) -> *mut Isolate {
+    crate::V8::assert_initialized();
+    let (raw_create_params, create_param_allocations) = params.finalize();
+    let cxx_isolate = unsafe { v8__Isolate__New(&raw_create_params) };
+    let isolate = unsafe { &mut *cxx_isolate };
+    isolate.initialize(create_param_allocations);
+    cxx_isolate
+  }
+
+  pub(crate) fn initialize(&mut self, create_param_allocations: Box<dyn Any>) {
```

Comment:
This method now handles general isolate initialization shared by snapshots, unique owned isolates, and shared isolates.

Previously the embedder data slots check was not done on snapshot isolates, which seemed odd and unintentional.

Reply:
This is probably worth splitting out into a separate PR as it seems reasonable on its own.

Reply:
I split the construction and disposal changes into https://github.com/denoland/rusty_v8/pull/1442